libs/header/p2p/exchange.go

```diff
@@ -235,18 +236,21 @@ func (ex *Exchange[H]) performRequest(
 		return nil, fmt.Errorf("no trusted peers")
 	}
 	var reqErr error
-	for i := 0; i < len(ex.trustedPeers); i++ {
-		//nolint:gosec // G404: Use of weak random number generator
-		idx := rand.Intn(len(ex.trustedPeers))
+	peers := ex.trustedPeers
 
+	// we need this each time before calling shuffle
+	rand.Seed(time.Now().UnixNano())
```

Comment:
Why do we need it each time? The seed should be populated once at the start of the app. The results of the shuffle will be different

Reply:
```
// Seed uses the provided seed value to initialize the default Source to a
// deterministic state. If Seed is not called, the generator behaves as
// if seeded by Seed(1). Seed values that have the same remainder when
// divided by 2³¹-1 generate the same pseudo-random sequence.
// Seed, unlike the Rand.Seed method, is safe for concurrent use.
```

Reply:
>  If Seed is not called, the generator behaves as if seeded by Seed(1)

I didn't say it should not be called, but I said it should be called only once when the node starts.

Reply:
But this is the comment in case we go with global randomness. The @walldiss's suggestion to create an instance each time as it soon be deprecated is also valid